cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -946,50 +1008,57 @@ func CreateSriovNetwork(sriovDevice *sriovv1.InterfaceExt, sriovNetworkName stri
 	}, time.Minute, 5*time.Second).ShouldNot(HaveOccurred())
 }
 
-func createDPDKWorkload(nodeSelector map[string]string, dpdkResourceName, testpmdCommand string, runningTime int, isServer bool) (*corev1.Pod, error) {
+func dpdkWorkloadCommand(dpdkResourceName, testpmdCommand string, runningTime int) string {
+	return fmt.Sprintf(`set -ex
+	export CPU=$(cat /sys/fs/cgroup/cpuset/cpuset.cpus)
+	echo ${CPU}
+	echo ${PCIDEVICE_OPENSHIFT_IO_%s}
+	
+	cat <<EOF >test.sh
+	spawn %s
+	set timeout 10000
+	expect "testpmd>"
+	send -- "port stop 0\r"
+	expect "testpmd>"
+	send -- "port detach 0\r"
+	expect "testpmd>"
+	send -- "port attach ${PCIDEVICE_OPENSHIFT_IO_%s}\r"
+	expect "testpmd>"
+	send -- "port start 0\r"
+	expect "testpmd>"
+	send -- "start\r"
+	expect "testpmd>"
+	sleep %d
+	send -- "stop\r"
+	expect "testpmd>"
+	send -- "quit\r"
+	expect eof
+	EOF
+	
+	expect -f test.sh
+	
+	sleep INF
+	`, dpdkResourceName, testpmdCommand, dpdkResourceName, runningTime)
+}
+
+func createDPDKWorkload(nodeSelector map[string]string, command string, isServer bool) (*corev1.Pod, error) {
 	resources := map[corev1.ResourceName]resource.Quantity{
 		corev1.ResourceName("hugepages-1Gi"): resource.MustParse("2Gi"),
 		corev1.ResourceMemory:                resource.MustParse("1Gi"),
 		corev1.ResourceCPU:                   resource.MustParse("4"),
 	}
+	privileged := true
 	container := corev1.Container{
 		Name:  "dpdk",
 		Image: images.For(images.Dpdk),
 		Command: []string{
 			"/bin/bash",
 			"-c",
-			fmt.Sprintf(`set -ex
-export CPU=$(cat /sys/fs/cgroup/cpuset/cpuset.cpus)
-echo ${CPU}
-echo ${PCIDEVICE_OPENSHIFT_IO_%s}
-
-cat <<EOF >test.sh
-spawn %s
-set timeout 10000
-expect "testpmd>"
-send -- "port stop 0\r"
-expect "testpmd>"
-send -- "port detach 0\r"
-expect "testpmd>"
-send -- "port attach ${PCIDEVICE_OPENSHIFT_IO_%s}\r"
-expect "testpmd>"
-send -- "port start 0\r"
-expect "testpmd>"
-send -- "start\r"
-expect "testpmd>"
-sleep %d
-send -- "stop\r"
-expect "testpmd>"
-send -- "quit\r"
-expect eof
-EOF
-
-expect -f test.sh
-
-sleep INF
-`, dpdkResourceName, testpmdCommand, dpdkResourceName, runningTime)},
+			command,
+		},
 		SecurityContext: &corev1.SecurityContext{
-			RunAsUser: pointer.Int64Ptr(0),
+			Privileged: &privileged, // TODO: remove
```

Comment:
we should not use privileged here.

I think using `NET_ADMIN` and `NET_RAW` should be ok

Reply:
We need NET_RAW, NET_ADMIN and MKNOD (to set up a tap device).
Removed privileged